### PR TITLE
gather_data: collect more job data to test bucket

### LINKED ISSUE #52: Don't open files earlier then `since`
Put this code to `gather_job_data.py`:

```
            if self.since_seconds:
                time = github_time_to_unix(log_file_list[0].split(
                    ' ', 1)[0].split('.', 1)[0])
                if curr_time - time > self.since_seconds:
                    self.gathered_test_data.pop(log_id)
                    break
```

### LINKED ISSUE #55: Merge gather_test_data and gather_job_data
There's a lot of reused code in these two files.

Start with readme.md & docs, write the new CLI with examples.

### PR DESCRIPTION
**Pull request by lastoCHka42**
Collect the following data:
* test type (tap, luatest, diff)
* diff test subtype (sql, python)
* runner architecture
* gc64/nogc64
* os distro and version
* debug or release build

Resolves https://github.com/tarantool/multivac/issues/51

### CODE REVIEW

File: multivac/gather_data.py [hunk 1]
```diff
@@ -130,14 +130,36 @@ def get_test_data(log_file) -> list:
...
                            'test_attempt': test_attempt}
             test_attempt = 1
             tests_data.append(test_record)
 
         return tests_data
 
+    @staticmethod
+    def get_release_or_debug(log_file):
+        for line in log_file:
+            if 'Run make -f' in line and line.endswith('debug\n'):
```
Comment (ylobankov) [inline multivac/gather_data.py:159]: I am not sure that is the right way to get `release` or `debug` build. You rely on make target name here, which is fragile. One day we can rename these targets and your logic will be broken.

File: multivac/gather_data.py [hunk 2]
```diff
@@ -197,6 +219,7 @@ def gather_data(self):
                     log_file_as_list = list(log_file)
                     time_queued = log_file_as_list[0][0:19] + 'Z'
                     test_data = self.get_test_data(log_file_as_list)
+                    debug = self.get_release_or_debug(log_file_as_list)
```
Comment (NickVolynkin) [inline multivac/gather_data.py:222]: static method shouldn't require `self`

File: multivac/gather_data.py [hunk 3]
```diff
@@ -317,13 +341,20 @@ def put_to_db_test(self):
...
+                    'test_type': test['test_type'],
+                    'diff_test_subtype': test['subtype'],
+                    'debug': job_info['debug'],
+                    'job_id': job_id,
+                    'job_name': job_info['job_name'],
+                    'commit_sha': job_info['commit_sha'],
+                    'test_attempt': test['test_attempt'],
+                    'architecture': job_info['platform'],
+                    'gc64': job_info['gc64'],
+                    'os_version': job_info['runner_label'][0],
```
Comment (NickVolynkin) [inline multivac/gather_data.py:357]: are you sure that the first label is always the OS?

**Review by NickVolynkin — APPROVED** [review]

File: multivac/gather_data.py (hunk 3 above)
Comment (lastoCHka42) [reply multivac/gather_data.py:358]: There are always only one label, as I can see

File: multivac/gather_data.py [hunk 4]
```diff
@@ -317,13 +341,20 @@ def put_to_db_test(self):
...
+            job_info = self.gathered_data[job_id]
+            for test in job_info.get('failed_tests'):
                 tags = {
-                    'job_id': job_id,
                     'configuration': test['conf'],
-                    'job_name': self.gathered_data[job_id]['job_name'],
-                    'commit_sha': self.gathered_data[job_id]['commit_sha'],
-                    'test_attempt': test['test_attempt']
+                    'test_type': test['test_type'],
+                    'diff_test_subtype': test['subtype'],
```
[the remark this answers is not in the document]
Comment (lastoCHka42) [reply multivac/gather_data.py:350]: Done

File: multivac/gather_data.py (hunk 1 above)
Comment (lastoCHka42) [reply multivac/gather_data.py]: Fixed

File: multivac/gather_data.py [hunk 5]
```diff
@@ -130,14 +130,37 @@ def get_test_data(log_file) -> list:
...
+            else:
+                test_type = 'diff'
+                if test_type.endswith('py'):
+                    subtype = 'python'
+                elif 'sql' in test_type:
+                    subtype = 'sql'
             test_record = {'name': test_name,
                            'conf': conf or 'none',
+                           'test_type': test_type,
+                           'subtype': subtype,
```
Comment (ylobankov) [inline multivac/gather_data.py:149]: Shouldn't we use `test_subtype` everywhere for consistency?

**Review by ylobankov — APPROVED** [review]
LGTM

**Pull request merged**